### PR TITLE
Rec: offensive language

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Deprecate uses of master, white- and blacklist in settings, providing alternative names. Old names are still accepted and processed.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [ ] tested this code
- [X] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: pdns/pdns_recursor.cc [hunk 1]
```diff
@@ -4228,6 +4228,7 @@ static void setupNODGlobal()
   g_nodLookupDomain = DNSName(::arg()["new-domain-lookup"]);
   g_nodLog = ::arg().mustDo("new-domain-log");
   parseNODWhitelist(::arg()["new-domain-whitelist"]);
+  parseNODWhitelist(::arg()["new-domain-ignore-list"]);
```
Comment (rgacogne) [inline pdns/pdns_recursor.cc:4231]: We should like change the function names as well :+1:

Comment (omoerbeek) [reply pdns/pdns_recursor.cc:4231]: > We should like change the function names as well 👍

Agreed, I concentrated on the externally visible words first.

**Comment by omoerbeek** [comment]
rpzMaster (in Lua config) shoud be taken care of as well.

File: pdns/rec-lua-conf.cc [hunk 2]
```diff
@@ -191,6 +191,127 @@ static void parseFrameStreamOptions(boost::optional<frameStreamOptions_t> vars,
...
+        tt.name=DNSName(toLower(boost::get<string>(have["tsigname"])));
+        tt.algo=DNSName(toLower(boost::get<string>(have[ "tsigalgo"])));
+        if(B64Decode(boost::get<string>(have[ "tsigsecret"]), tt.secret))
+          throw std::runtime_error("TSIG secret is not valid Base-64 encoded");
+      }
+
+      if(have.count("refresh")) {
+        refresh = boost::get<uint32_t>(have["refresh"]);
+        if (refresh == 0) {
+          g_log<<Logger::Warning<<"rpzMaster refresh value of 0 ignored"<<endl;
```
Comment (rgacogne) [inline pdns/rec-lua-conf.cc:246]:
```suggestion
          g_log<<Logger::Warning<<"rpzPrimary refresh value of 0 ignored"<<endl;
```

File: pdns/rec-lua-conf.cc [hunk 3]
```diff
@@ -191,6 +191,127 @@ static void parseFrameStreamOptions(boost::optional<frameStreamOptions_t> vars,
...
+      if(have.count("dumpFile")) {
+        dumpFile = boost::get<std::string>(have["dumpFile"]);
+      }
+    }
+
+    if (localAddress != ComboAddress()) {
+      // We were passed a localAddress, check if its AF matches the masters'
+      for (const auto& master : masters) {
+        if (localAddress.sin4.sin_family != master.sin4.sin_family) {
+          throw PDNSException("Master address("+master.toString()+") is not of the same Address Family as the local address ("+localAddress.toString()+").");
```
Comment (rgacogne) [inline pdns/rec-lua-conf.cc:275]:
```suggestion
          throw PDNSException("Primary address("+master.toString()+") is not of the same Address Family as the local address ("+localAddress.toString()+").");
```

File: pdns/rec-lua-conf.cc [hunk 4]
```diff
@@ -191,6 +191,127 @@ static void parseFrameStreamOptions(boost::optional<frameStreamOptions_t> vars,
...
+          throw PDNSException("The RPZ zone " + zoneName + " loaded from the seed file (" + zone->getDomain().toString() + ") has no SOA record");
+        }
+      }
+      catch(const std::exception& e) {
+        g_log<<Logger::Warning<<"Unable to pre-load RPZ zone "<<zoneName<<" from seed file '"<<seedFile<<"': "<<e.what()<<endl;
+      }
+    }
+  }
+  catch(const std::exception& e) {
+    g_log<<Logger::Error<<"Problem configuring 'rpzMaster': "<<e.what()<<endl;
```
Comment (rgacogne) [inline pdns/rec-lua-conf.cc:304]:
```suggestion
    g_log<<Logger::Error<<"Problem configuring 'rpzPrimary': "<<e.what()<<endl;
```

File: pdns/rec-lua-conf.cc [hunk 5]
```diff
@@ -191,6 +191,127 @@ static void parseFrameStreamOptions(boost::optional<frameStreamOptions_t> vars,
...
+    exit(1);  // FIXME proper exit code?
+  }
+  catch(const PDNSException& e) {
+    g_log<<Logger::Error<<"Problem configuring 'rpzMaster': "<<e.reason<<endl;
```
Comment (rgacogne) [inline pdns/rec-lua-conf.cc:308]:
```suggestion
    g_log<<Logger::Error<<"Problem configuring 'rpzPrimary': "<<e.reason<<endl;
```

**Review by rgacogne — APPROVED** [review]
A few nits but the PR looks great :+1:

**Comment by omoerbeek** [comment]
Rebased

**Pull request merged**